### PR TITLE
[NET Attribute Conversion] Rerun with fixes to catch old/unlisted availability

### PR DESCRIPTION
**Pull request by chamons**
- Turn on one of two tests from https://github.com/xamarin/xamarin-macios/pull/14106
- The other one, and the currently disabled bits will be enabled in a future PR that corrects the generator.

This re-run of of mellite was powered by two changes:

- https://github.com/chamons/mellite/commit/650f58479c9cecfa2377a39f68283106a6404f46 which introduced `add-default-introduced` that will detect when an API is available on a given platform without availability attributes and create a default "non-versioned" one.
- A hack of XM and XI dll which was built with https://github.com/xamarin/xamarin-macios/blob/main/src/generator.cs#L3368 set to never filter any attributes. This way, attributes that existed but were "too old" were there for mel to read.

After that, I could re-run it against all of xamarin-macios (repeating the annoying one time work), and then fixup the test to work.

### CODE REVIEW

**Comment by chamons** [comment]
Further work on the generator suggests this PR may be incomplete (due to attributes not being seen).

Going to verify before closing or adding to this.

File: tests/cecil-tests/AttributeTest.cs [hunk 1]
```diff
@@ -37,7 +37,7 @@ public class AttributeTest {
 		//
 		// This test should find Extension, note that it has an ios attribute,
 		// and insist that some maccatalyst must also be set.
-		// [TestCaseSource (typeof (Helper), "NetPlatformAssemblies")]
+		[TestCaseSource (typeof (Helper), "NetPlatformAssemblies")]
```
Comment (rolfbjarne) [inline tests/cecil-tests/AttributeTest.cs:40]: This avoids a string constant:

```suggestion
		[TestCaseSource (typeof (Helper), nameof (Helper.NetPlatformAssemblies))]
```

File: tests/cecil-tests/AttributeTest.cs [hunk 2]
```diff
@@ -179,7 +179,27 @@ void DebugPrint (string fullName, IEnumerable<string> parentAvailability, IEnume
 
 		bool Ignore (string fullName)
 		{
+			string ns;
+			if (fullName.Split(' ').Length < 2) {
+				ns = fullName.Split('.').First ();
```
Comment (rolfbjarne) [inline tests/cecil-tests/AttributeTest.cs:184]:
```suggestion
			if (fullName.Split (' ').Length < 2) {
				ns = fullName.Split ('.').First ();
```

File: tests/cecil-tests/AttributeTest.cs [hunk 3]
```diff
@@ -179,7 +179,27 @@ void DebugPrint (string fullName, IEnumerable<string> parentAvailability, IEnume
...
+			}
+			else {
+				// Skip the return type, then split namespace off
+				ns = fullName.Split(' ')[1].Split('.').First ();
```
Comment (rolfbjarne) [inline tests/cecil-tests/AttributeTest.cs:188]:
```suggestion
				ns = fullName.Split (' ') [1].Split ('.').First ();
```

File: tests/cecil-tests/AttributeTest.cs [hunk 4]
```diff
@@ -179,7 +179,27 @@ void DebugPrint (string fullName, IEnumerable<string> parentAvailability, IEnume
 
 		bool Ignore (string fullName)
 		{
+			string ns;
+			if (fullName.Split(' ').Length < 2) {
+				ns = fullName.Split('.').First ();
+			}
+			else {
```
Comment (rolfbjarne) [inline tests/cecil-tests/AttributeTest.cs:186]:
```suggestion
			} else {
```

File: tests/cecil-tests/AttributeTest.cs [hunk 5]
```diff
@@ -179,7 +179,27 @@ void DebugPrint (string fullName, IEnumerable<string> parentAvailability, IEnume
...
+				// Skip the return type, then split namespace off
+				ns = fullName.Split(' ')[1].Split('.').First ();
+			}
+			switch (ns) {
+				case "Security":
+				case "CoreMedia":
+				case "CoreText":
+					return true;
```
Comment (rolfbjarne) [inline tests/cecil-tests/AttributeTest.cs:194]:
```suggestion
			case "Security":
			case "CoreMedia":
			case "CoreText":
				return true;
```

**Review by rolfbjarne — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by chamons** [comment]
Re-ran converter with new missing attribute logic to detect tv/catalyst attributes and use "what exists in the assembly" not "What namespace are you in".

It's even larger, but should only be supported/unsupported attributes.

I'm going to test this with the generator changes in flight and see if it checks out finally...

File: src/UIKit/UIAccessibility.cs [hunk 6]
```diff
@@ -300,18 +326,24 @@ static unsafe void TrampolineRequestGuidedAccessSession (IntPtr block, bool enab
 			}
 		}
 
-#if !NET
+#if NET
```
Comment (rolfbjarne) [inline src/UIKit/UIAccessibility.cs:329]: This is re-introducing a member to .NET that we don't want (the `#if !NET` surrounded the entire property before the change, but after the change only the availability attributes).

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/UIKit/UIAccessibility.cs (hunk 6 above)
Comment (chamons) [reply src/UIKit/UIAccessibility.cs:329]: Good catch.

File: src/AVFoundation/AVCaptureConnection.cs [hunk 7]
```diff
@@ -39,6 +39,8 @@ public partial class AVCaptureConnection {
 
 #if NET
 		[SupportedOSPlatform ("maccatalyst14.0")]
+		[SupportedOSPlatform ("ios")]
+		[SupportedOSPlatform ("macos")]
```
Comment (mandel-macaque) [inline src/AVFoundation/AVCaptureConnection.cs:43]: This has an obsolete on iOS, should it have one in maccatalyst?

File: src/AVFoundation/AVPlayerViewController.cs [hunk 8]
```diff
@@ -21,6 +21,8 @@ public partial class AVPlayerViewController {
 		// note: we cannot reuse the same method name - as it would break compilation of existing apps
 #if NET
 		[SupportedOSPlatform ("ios8.0")]
+		[SupportedOSPlatform ("maccatalyst")]
+		[SupportedOSPlatform ("tvos")]
```
Comment (mandel-macaque) [inline src/AVFoundation/AVPlayerViewController.cs:25]: Same

**Review by mandel-macaque — CHANGES_REQUESTED** [review]
The issue I see is as follows, we have obsoleted APIs in iOS exposed in MacCa1talyst that are missing the MacCatalyst Obsoleted OR they show Obsoleted with the iOS version (since there is no #if MACCATALYST we default to the iOS message AFAIK.

**Comment by chamons** [comment]
@mandel-macaque - I filed https://github.com/xamarin/xamarin-macios/issues/14178 to look at those deprecated outside of this PR.

**Comment by chamons** [comment]
I was able to run Cecil.Tests.AttributeTest.ChildElementsListAvailabilityForAllPlatformsOnParent with this diff and a few test changes, which suggests the API changes are good.

**Review by dalexsoto — APPROVED** [review]
Could not find anything else, great stuff 👍

**Review by rolfbjarne — APPROVED** [review]

**Comment by chamons** [comment]
Looking into test failures now that it's passing and the one test last night.

**Comment by chamons** [comment]
This required a lot of work to fix and improve on, so I'm closing and opening a new PR.

**Pull request closed without merging**